File: Examples/VisualBasic/Loading-and-Saving/CreateDocWithSimpleRichText.vb
Imports System.IO
Imports AsposeNote = Aspose.Note
Imports System.Drawing
Namespace Loading_Saving
    Public Class CreateDocWithSimpleRichText
        Public Shared Sub Run()
            ' ExStart:CreateDocWithSimpleRichText
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_LoadingAndSaving()

            ' Create an object of the Document class
            Dim doc As New AsposeNote.Document()
            ' Initialize Page class object
            Dim page As New AsposeNote.Page(doc)
            ' Initialize Outline class object
            Dim outline As New Outline(doc)
            ' Initialize OutlineElement class object
            Dim outlineElem As New OutlineElement(doc)

            ' Initialize TextStyle class object and set formatting properties
            Dim textStyle As New TextStyle() With {.FontColor = Color.Black, .FontName = "Arial", .FontSize = 10}
            ' Initialize RichText class object and apply text style
            Dim text As New RichText(doc) With {.Text = "Hello OneNote text!", .DefaultStyle = textStyle}

            ' Add RichText node
            outlineElem.AppendChild(text)
            ' Add OutlineElement node
            outline.AppendChild(outlineElem)
            ' Add Outline node
            page.AppendChild(outline)
            ' Add Page node
            doc.AppendChild(page)

            dataDir = dataDir + "CreateDocWithSimpleRichText_out_.one"
            ' Save OneNote document
            doc.Save(dataDir)
            ' ExEnd:CreateDocWithSimpleRichText
            Console.WriteLine(Convert.ToString(vbLf & "OneNote document created successfully with simple rich text." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Loading-and-Saving/CreateDocWithSimpleRichText.cs
using System.IO;
using Aspose.Note;
using Aspose.Note.Saving;
using System;
using System.Drawing;

namespace Aspose.Note.Examples.CSharp.Loading_Saving
{
    public class CreateDocWithSimpleRichText
    {
        public static void Run()
        {
            // ExStart:CreateDocWithSimpleRichText
            // ExFor:Document
            // ExFor:RichText
            // ExFor:RichText.Text
            // ExSummary:Shows how to create a document with a text.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_LoadingAndSaving();

            // Create an object of the Document class
            Document doc = new Document();

            // Initialize Page class object
            Page page = new Page(doc);
            
            // Initialize Outline class object
            Outline outline = new Outline(doc);
            
            // Initialize OutlineElement class object
            OutlineElement outlineElem = new OutlineElement(doc);

            // Initialize TextStyle class object and set formatting properties
            ParagraphStyle textStyle = new ParagraphStyle { FontColor = Color.Black, FontName = "Arial", FontSize = 10 };
            
            // Initialize RichText class object and apply text style
            RichText text = new RichText(doc) { Text = "Hello OneNote text!", ParagraphStyle = textStyle };

            // Add RichText node
            outlineElem.AppendChildLast(text);
            
            // Add OutlineElement node
            outline.AppendChildLast(outlineElem);
            
            // Add Outline node
            page.AppendChildLast(outline);
            
            // Add Page node
            doc.AppendChildLast(page);

            // Save OneNote document
            dataDir = dataDir + "CreateDocWithSimpleRichText_out.one";
            doc.Save(dataDir);
            
            // ExEnd:CreateDocWithSimpleRichText

            Console.WriteLine("\nOneNote document created successfully with simple rich text.\nFile saved at " + dataDir);
        }
    }
}
